Source/Core/VideoBackends/Vulkan/VKSwapChain.cpp

```diff
@@ -36,8 +36,173 @@ SwapChain::~SwapChain()
+      return VK_NULL_HANDLE;
+    }
+
+    if (mode_count == 0)
+    {
+      ERROR_LOG_FMT(VIDEO, "Cannot find any mode for the display!");
+      return VK_NULL_HANDLE;
+    }
+
+    VkDisplayModePropertiesKHR mode_props;
+    mode_count = 1;
+    if (VkResult err = vkGetDisplayModePropertiesKHR(physical_device, display_props.display,
+                                                     &mode_count, &mode_props);
+        err != VK_SUCCESS && err != VK_INCOMPLETE)
+    {
+      LOG_VULKAN_ERROR(err, "vkGetDisplayModePropertiesKHR failed: ");
+      return VK_NULL_HANDLE;
+    }
+
+    // Get the list of planes
+    uint32_t plane_count = 0;
+    if (VkResult err =
+            vkGetPhysicalDeviceDisplayPlanePropertiesKHR(physical_device, &plane_count, nullptr);
+        err != VK_SUCCESS)
+    {
+      LOG_VULKAN_ERROR(err, "vkGetPhysicalDeviceDisplayPlanePropertiesKHR failed: ");
+      return VK_NULL_HANDLE;
+    }
+
+    if (plane_count == 0)
+    {
+      ERROR_LOG_FMT(VIDEO, "No display planes found!");
+      return VK_NULL_HANDLE;
+    }
+
+    // Find a plane compatible with the display
+    uint32_t compatible_plane_index = UINT32_MAX;
+
+    for (uint32_t plane_index = 0; plane_index < plane_count; plane_index++)
+    {
+      // Query the number of displays supported by the plane
+      display_count = 0;
+      VkResult err = vkGetDisplayPlaneSupportedDisplaysKHR(physical_device, plane_index,
+                                                           &display_count, nullptr);
+      if (err != VK_SUCCESS)
+      {
+        LOG_VULKAN_ERROR(err, "vkGetDisplayPlaneSupportedDisplaysKHR (count query) failed: ");
+        return VK_NULL_HANDLE;
+      }
+
+      if (display_count == 0)
+        continue;  // Skip planes that support no displays
+
+      // Allocate memory to hold the supported displays
+      std::vector<VkDisplayKHR> displays(display_count);
+      err = vkGetDisplayPlaneSupportedDisplaysKHR(physical_device, plane_index, &display_count,
+                                                  displays.data());
+      if (err != VK_SUCCESS)
+      {
+        LOG_VULKAN_ERROR(err, "vkGetDisplayPlaneSupportedDisplaysKHR (fetch displays) failed: ");
+        return VK_NULL_HANDLE;
+      }
+
+      // Check if the target display is among the supported displays
+      for (const auto& display : displays)
+      {
+        if (display == display_props.display)
+        {
+          compatible_plane_index = plane_index;
+          break;
+        }
+      }
+
+      if (compatible_plane_index != UINT32_MAX)
+        break;  // Exit early if a compatible plane is found
+    }
+
+    if (compatible_plane_index == UINT32_MAX)
+    {
+      ERROR_LOG_FMT(VIDEO, "No compatible plane found for the display!");
+      return VK_NULL_HANDLE;
+    }
+
+    if (compatible_plane_index == UINT32_MAX)
+    {
+      ERROR_LOG_FMT(VIDEO, "No compatible plane found for the display!");
+      return VK_NULL_HANDLE;
+    }
```

Comment:
```suggestion
    if (compatible_plane_index == UINT32_MAX)
    {
      ERROR_LOG_FMT(VIDEO, "No compatible plane found for the display!");
      return VK_NULL_HANDLE;
    }
```

Same code is repeated twice in a row.